File: gto-support-jsonapi/src/androidTest/java/org/ccci/gto/android/common/jsonapi/model/ModelSimple.java
package org.ccci.gto.android.common.jsonapi.model;

import org.ccci.gto.android.common.jsonapi.annotation.JsonApiType;

@JsonApiType(value = ModelSimple.TYPE, aliases = {ModelSimple.ALIAS1, ModelSimple.ALIAS2})
public final class ModelSimple extends ModelBase {
    public static final String TYPE = "simple";
    public static final String ALIAS1 = "aliased";
    public static final String ALIAS2 = "simple_alias";
    public static final String NOTALIAS = "notsimple";

    public ModelSimple() {}

    public ModelSimple(final int id) {
        mId = id;
    }
}

File: gto-support-jsonapi-core/src/test/kotlin/org/ccci/gto/android/common/jsonapi/model/ModelSimple.kt
package org.ccci.gto.android.common.jsonapi.model

import org.ccci.gto.android.common.jsonapi.annotation.JsonApiType

@JsonApiType(value = ModelSimple.TYPE, aliases = [ModelSimple.ALIAS1, ModelSimple.ALIAS2])
class ModelSimple(id: Int? = null) : ModelBase(id) {
    companion object {
        const val TYPE = "simple"
        const val ALIAS1 = "aliased"
        const val ALIAS2 = "simple_alias"
        const val NOTALIAS = "notsimple"
    }
}
